Updating Ubuntu packages and expanding Docker sections

Udinanon opened the pull request:
I expanded the packages suggested for ubuntu/debian to reflect the ones i needed in Docker. Some might already be installed like the python3-minimal, but that should just ensure that the user machine is not missing any components I also expanded the Docker build sections to include infos on Installing Docker and a bit more background on how it works.

Udinanon committed:
Updating Ubuntu packages and expanding Docker sections

I expanded the packages suggested for ubuntu/debian to reflect the ones i needed in Docker.
Some might already be installed like the python3-minimal, but that should just ensure that the user machine is not missing any components
I also expanded the Docker build sections to include infos on Installing Docker and a bit more background on how it works.

bjornbytes commented on guides/Compiling.md:
```diff
@@ -79,11 +79,20 @@ using one of those builds.
 ### Debian/Ubuntu
 
-    $ sudo apt install cmake clang xorg-dev libcurl-dev
+    $ sudo apt install cmake clang xorg-dev build-essential libcurl4-openssl-dev libxcb-glx0-dev libx11-xcb-dev glslang-tools python3-minimal
```
- `glslang` does not need to be installed to build LÖVR for Linux.  Since we have to build glslang from source anyway, we use that to compile shaders.  Android is another story, but that's already documented in the Android section.
- I don't think it makes sense to include both `build-essential` and `clang` since this installs 2 different C compilers.
- Yes, we should change `libcurl-dev` to `libcurl4-openssl-dev`, since `libcurl-dev` is a virtual package.
- Since python isn't required to build anything directly in LÖVR and it's already installed by clang, I think it would be better to leave it out.

Udinanon replied:
I think python is needed somewhere if you build it from scratch, i think something in Vulkan? i can't build it in docker unless i install it
`clang` i just took from the docs, but build essentials does something to inform CMake of the available compilers, maybe this can be done in some other way
I need to check with `glslang` cause it didn't work without it I think, at least in docker

Maybe the differences come from a full system VS what the image has?

bjornbytes replied:
Hm maybe if we only install clang then we are missing a C++ compiler (build-essential installs g++), I'll try it on a fresh Ubuntu.

Udinanon replied:
Ok `build-essentials` can be skipped if we set the `CC` and `CXX` variables, that can be done in the dockerfile, we only need to also install `make`
python3 is required at cpmpilation by glslang actually
```
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Python3 (missing: Python3_EXECUTABLE Interpreter)
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.22/Modules/FindPython/Support.cmake:3180 (find_package_handle_standard_args)
  /usr/share/cmake-3.22/Modules/FindPython3.cmake:490 (include)
  deps/glslang/CMakeLists.txt:197 (find_package)
  deps/glslang/StandAlone/CMakeLists.txt:34 (find_host_package)


-- Configuring incomplete, errors occurred!
```

glslang does not need to be isntalled, I'll remove it

bjornbytes commented:
If we have 2 identical Docker sections, there is a risk that they'll get out of sync.  We should probably have a top-level Docker section (a `---` header alongside Android, Linux, etc.) that describes which host platforms it supports and which target platforms it can build for.

Udinanon commented:
The larger section might be a better design, I'm actually looking into if https://github.com/dockur/windows can be used to also build for windows with less work than other QEMU solutions

Udinanon committed:
Moved into independent section

I hope i'm using the patch thing correctly
I moved the Using Docker to a section
Removed unneeded packages like `glslang` from the ubuntu instructions
I don't think we want to duplicate the instructions here? it's just more work if they change and they will probably have to 
I think python3 is needed, maybe the python3-minimal is not the best package? but glslang wants some version and this ensures one

bjornbytes commented:
Sweet, ty!
